daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala

```diff
@@ -246,7 +248,7 @@ class Engine(val config: EngineConfig = new EngineConfig(LanguageVersion.StableV
         case speedy.Compiler.PackageNotFound(_) =>
           handleMissingDependencies.flatMap(_ => start)
         case speedy.Compiler.CompilationError(error) =>
-          ResultError(Error.Preprocessing.Generic(s"CompilationError: $error"))
+          ResultError(Error.Preprocessing.Internal(funcName, s"CompilationError: $error"))
```

Comment:
maybe add a comment why this internal. Presumably because typechecking should have failed earlier?

Reply:
:+1: